src/liballoc/boxed.rs

```diff
@@ -4,6 +4,16 @@
 //! heap allocation in Rust. Boxes provide ownership for this allocation, and
 //! drop their contents when they go out of scope.
 //!
+//! A [`Box`] will use the [`Global`] allocator for its allocation. It is valid
```

Comment:
Is this all true for zero-sized types?

Reply:
Ah yeah, we need to add an exception for ZSTs since those can't interact with global allocators.

Reply:
Added qualifier.

Reply:
Is there a reason `std::alloc::Layout::from_size_align(0, 1)` returns `Ok`?

Reply:
Creating a (possibly) zero-size layout is valid. It could be used as an intermediate value passed to `Layout::extend`, for example.